test/Test/Toml/Printer/Golden.hs

```diff
@@ -34,17 +35,28 @@ example = mkToml $ do
 
 noFormatting :: PrintOptions
 noFormatting = PrintOptions
-    { shouldSort = False
-    , indent     = 0
+    { printOptionsSorting = Nothing
+    , printOptionsIndent  = 0
     }
 
+-- | Decorate keys as tuples so spam comes before egg
+spamEggDecorate :: Key -> (Int, Key)
+spamEggDecorate k
+    | k == "spam" = (0, "spam")
+    | k == "egg" = (1, "egg")
+    | otherwise = (2, k)
+
+spamEgg :: Key -> Key -> Ordering
+spamEgg = comparing spamEggDecorate
```

Comment:
Maybe a slightly more simple example of defining a custom ordering could be `spamEgg = compare on length` or something like that.

Reply:
Realistic use-cases of this feature involve key orders chosen semantically to serve human comprehension. They are usually arbitrary with respect to the key text. I'd like there to be some advertisement of how to to pretty-print a textually arbitrary key order. I think tuple decoration is a straightforward way to do so. This test is the only such advertisement so far. I'd personally have trouble reverse-engineering `compare \`on\` length` to design a semantically relevant sorter. I suffered an extreme amount of this type of obscurity among the Haskell YAML options, finally migrating a project to TOML because `tomland`'s codebase and docs are actually comprehensible. This type of comprehensibility matters a lot to me.

If it's important to keep the test code concise, we can go this way. But if so, let's discuss where to present a realistic pattern for a custom sort, either in the package documentation or in the blogosphere.

Reply:
I often use the pattern ``compare `on` f`` so to me the meaning is super clear: sort by string length. I understand if you are less used to it and you prefer more explicit, semantic functions, it's OK to do that. 

However, I feel a bit uneasy about the signature `spamEggDecorate :: Key -> (Int, Key)`. You are doing a function to sort `Key`, so you transform it to an `Int` but you still keep the `Key`, if two keys have the same integer value, you default to alphabetical order. I guess it's not such a big deal, but maybe `spamEggDecorate :: Key -> Int` would be more clear? It's up to you.

I can't really think of an ordering I'd want more than plain alphabetical to be honest, so I'll leave it up to you. As far as I'm concerned, the current PR is great.

Reply:
I grant that `(Int, Key)` has more cardinality than actually used in the sort.

The common use-case that motivated this feature is when one field in an array of tables/objects is critical for comprehension but gets buried in an alphabetical pretty-printing. Using an example from the `toml` readme, an alphabetically sorted array of tables might look like this:

```
[[products]]
name = "Hammer"
sku = 738594937

[[products]]
color = "gray"
name = "Nail"
sku = 284758393
```

Putting the name up front makes the table easier for a human to scan:

```
[[products]]
name = "Hammer"
sku = 738594937

[[products]]
name = "Nail"
color = "gray"
sku = 284758393
```

Realistic data structures tend to suffer from a more extreme version of this problem because they tend to be bigger, often with substantial nesting; I've found alphabetically pretty-printed data structures over the course of my career surprisingly hard to read. I frequently want "alphabetical with exceptions" to make pretty-printed data more legible, and the given sorter implements this.

For the purpose of the test suite, the given sorter also corresponds to a minimal change in the golden file (transposing two lines of TOML compared to the alphabetical sort, which I was able to construct by hand), helping to verify correctness of the library code.

Reply:
I see, you make a good case. For the purpose of wanting to get `name` on top, `(Int, Key)` is actually nice since anything non-covered explicitly will be alphabetical sorted. 

Thank you for sharing your insights on realistic uses of config files, that is helpful to me.